lib/season_statistics.rb

```diff
@@ -0,0 +1,119 @@
+require 'pry'
+
+class SeasonStatistics
+attr_reader :stat_tracker, :game_collection, :game_teams_collection, :teams_collection
+
+  def initialize(game_collection, game_teams_collection, teams_collection)
+    @game_collection = game_collection
+    @game_teams_collection = game_teams_collection
+    @teams_collection = teams_collection
+  end
+
+  def current_season_games(season)
+    current_games = @game_collection.map do |game|
+     if game.season == season
+       game.game_id
+     end
+   end
+   current_games.compact
+  end
+
+  def teams_hash(season)
+    teams = Hash.new(0)
+    season_teams = current_season_game_teams(season).map do |game|
+      game.team_id
+      end
+    @teams_collection.each do |team|
+        if season_teams.include?(team.id)
+          teams[team.id] = 0
+        end
+      end
+    teams
+  end
+
+  def coaches_hash(season)
+    coaches = Hash.new(0)
+    season_coaches = current_season_game_teams(season).map do |game|
+      game.head_coach
+      end
+    @game_teams_collection.each do |team|
+        if season_coaches.include?(team.head_coach)
+          coaches[team.head_coach] = 0
+        end
+      end
+    coaches
+  end
+
+  def current_season_game_teams(season)
+    current = @game_teams_collection.find_all do |game|
+    current_season_games(season).include?(game.game_id)
```

Comment:
I think this is a possible source of the performance issue. You are calling the current_season_games method for every game in the game_teams_collection. Try to call it once and save it to a variable.

Reply:
it would have to be saved to an instance variable right?